src/collection.rs

```diff
@@ -0,0 +1,450 @@
+use std::borrow::Cow;
+
+/// Generic collection of an abstracted item.
+///
+/// This is used to abstract over the different types of collections that can be used,
+/// such as `Vec<T>`, `&[T]`, `&[&T]`, `&[&str]`, etc.
+///
+/// In most cases, you can use the [`Collection::from`] method to create a collection.
+/// If you get an error
+///
+/// # Examples
+///
+/// ```rust
+/// use twitch_types::{Collection, UserId, UserIdRef};
+///
+/// // A vector of `UserId`s
+/// let c0: Collection<UserId> = Collection::from(vec![UserId::from("1234"), UserId::from("5678")]);
+/// // A vector of `&str`s
+/// let c1: Collection<UserId> = Collection::from(vec!["1234", "5678"]);
+/// // An array of `&str`s
+/// let c2: Collection<UserId> = Collection::from(&["1234", "5678"]);
+/// // A vector of `UserIdRef`s
+/// let c3: Collection<UserId> = Collection::from(vec![
+///     UserIdRef::from_static("1234"),
+///     UserIdRef::from_static("5678"),
+/// ]);
+///
+/// assert!([c1, c2, c3].iter().all(|c| *c == c0));
+/// ```
+pub enum Collection<'c, T: std::ops::Deref + 'static>
```

Comment:
I wonder if the compiler could make this enum take just 24 bytes instead of 32 [on 64bit] if the index for the variant could use the padding of the index in `Cow` (which should be a single bit). Maybe I'm missing something though.

Reply:
padding isn't guaranteed to be zero, so it can't niche there

Reply:
could switch to another cow library that is smaller, but I'm not sure what would be lost with that, maybe worth it